tests/ui/use/use-path-segment-kw.rs

```diff
@@ -0,0 +1,177 @@
+//@ edition: 2021
+
+macro_rules! macro_dollar_crate {
+    ($m: ident) => {
+        use $crate; //~ ERROR `$crate` may not be imported
+        pub use $crate as _dollar_crate; // Good
+
+        use ::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use ::$crate as _dollar_crate2; //~ ERROR `$crate` in paths can only be used in start position
+        use ::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use ::{$crate as _nested_dollar_crate2}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use $m::$crate; //~ ERROR `$crate` in paths can only be used in start position
```

Comment:
```suggestion
        use foobar::$crate; //~ ERROR `$crate` in paths can only be used in start position
```
Will be simpler to read without passing it through `$m`.